tests/pipeline/test_pipeline.py

```diff
@@ -247,41 +246,41 @@ def test_with_custom_config_with_absolute_defaults_path(
-        abs_config_yaml = open(
-            str(RESOURCE_PATH / "custom-config/temp_config.yaml"), "w"
-        )
-        yaml.dump(config_dict, abs_config_yaml)
-        result = runner.invoke(
-            app,
-            [
-                "generate",
-                "--pipeline-base-dir",
-                PIPELINE_BASE_DIR,
-                str(RESOURCE_PATH / "custom-config/pipeline.yaml"),
-                "--config",
-                str(RESOURCE_PATH / "custom-config/temp_config.yaml"),
-            ],
-            catch_exceptions=False,
-        )
-
-        assert result.exit_code == 0
-
-        enriched_pipeline = yaml.safe_load(result.stdout)
-        assert isinstance(enriched_pipeline, dict)
-
-        producer_details = enriched_pipeline["components"][0]
-        output_topic = producer_details["app"]["streams"]["outputTopic"]
-        assert output_topic == "app1-test-topic"
-
-        streams_app_details = enriched_pipeline["components"][1]
-        output_topic = streams_app_details["app"]["streams"]["outputTopic"]
-        assert output_topic == "app2-test-topic"
-        error_topic = streams_app_details["app"]["streams"]["errorTopic"]
-        assert error_topic == "app2-dead-letter-topic"
-
-        snapshot.assert_match(enriched_pipeline, "test-pipeline")
-        abs_config_yaml.close()
-        os.remove(str(RESOURCE_PATH / "custom-config/temp_config.yaml"))
+        temp_config_path = RESOURCE_PATH / "custom-config/temp_config.yaml"
+        with open(str(temp_config_path), "w") as abs_config_yaml:
+            try:
+                yaml.dump(config_dict, abs_config_yaml)
+                result = runner.invoke(
+                    app,
+                    [
+                        "generate",
+                        "--pipeline-base-dir",
+                        PIPELINE_BASE_DIR,
+                        str(RESOURCE_PATH / "custom-config/pipeline.yaml"),
+                        "--config",
+                        str(temp_config_path),
+                    ],
+                    catch_exceptions=False,
+                )
+
+                assert result.exit_code == 0
+
+                enriched_pipeline = yaml.safe_load(result.stdout)
+                assert isinstance(enriched_pipeline, dict)
+
+                producer_details = enriched_pipeline["components"][0]
+                output_topic = producer_details["app"]["streams"]["outputTopic"]
+                assert output_topic == "app1-test-topic"
+
+                streams_app_details = enriched_pipeline["components"][1]
+                output_topic = streams_app_details["app"]["streams"]["outputTopic"]
+                assert output_topic == "app2-test-topic"
+                error_topic = streams_app_details["app"]["streams"]["errorTopic"]
+                assert error_topic == "app2-dead-letter-topic"
+
+                snapshot.assert_match(enriched_pipeline, "test-pipeline")
+            finally:
+                temp_config_path.unlink()
```

Comment:
Do you realy need a `try/finally` block here? can't you just call the unlink after the assert_match method?

Reply:
https://github.com/bakdata/kpops/pull/151#discussion_r1139163873

Reply:
I thought the same, but I didn't consider that the file would not be deleted under some circumstances. I would probably prefer to err on the safe side, who knows what might happen. It's a test anyway, it shouldn't impact kpops' performance, right?

Reply:
Okay that explains and answers my question. All good and fine! Thanks! 😄